### PR TITLE
Update ResourceIdentity reference and remove ManagedInstanceMetric

### PR DESCRIPTION
**Pull request by ericshape**
Most of the changes in this PR are  **auto-generated** by SQL DB DSMainDev Repo:

0. change the reference of ResourceIdentity in https://github.com/ericshape/azure-rest-api-specs/blob/2382a3b1c11562eeba18480e7b8fd5490ae0129f/specification/sql/resource-manager/Microsoft.Sql/preview/2019-06-01-preview/servers.json#L349-L376 **auto-generated**
1. add `None` and `UserAssigned` to ResourceIdentity in SQL specific common type file: https://github.com/ericshape/azure-rest-api-specs/blob/33b5733cc82028c9ce6e2c9ec9f28e7f763098f7/specification/sql/resource-manager/common/v1/types.json#L100-L129 **manually change**
    - It would lead to breaking change: Add new values to the enum.
2. update all the references in 2020-08-01-preview spec files.  **auto-generated**
3. remove all the ManagedInstanceMetric from 2020-08-01-preview, which is requested by the owner and not ready to publish in Public API.  **auto-generated**
    - It would lead to breaking change: removing paths. **But we have not published any doc and SDK for 2020-08-01-preview.** It is fine for us. 

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by jianyexi** [comment]
@PhoenixHe-msft  could you have a look at the failure of SDK azure-sdk-for-java

**Comment by ChenTanyi** [comment]
The Java automation is not blocking the PR merge. And it shows in log that the Java generation is successful.

**Comment by akning-ms** [comment]
Hi @ericschung, can I know when ManagedInstanceMetric is added? in which PR?

File: specification/sql/resource-manager/common/v1/types.json
Comment (akning-ms) [inline specification/sql/resource-manager/common/v1/types.json:119]: Can we keep this name? as it may cause sdk breaking change

**Comment by ericshape** [comment]
> Hi @ericschung, can I know when ManagedInstanceMetric is added? in which PR?

it was just added in 2020-08-01-preview two weeks ago. And we have not added it into any SDK and public doc. Thus, we are fine with it. 

BTW, you might tag the wrong ID LOL

File: specification/sql/resource-manager/common/v1/types.json
Comment (ericshape) [reply specification/sql/resource-manager/common/v1/types.json]: updated in the new iteration. feel free to check.

**Review by jamestao — APPROVED** [review]

**Comment by ericshape** [comment]
@akning-ms 
I have addressed your comments and @jamestao also approved this PR. For the breaking changes, I have explained them in the beginning of the PR description. 
Can you help us to merge this PR? Thanks!

BTW, the upcoming PR is ready: #11840 and it passed all the checks. Can you help to take a look? 
In #11840, I cherry-picked the changes from this PR #11960  to make it through. I will rebase it once #11960 is merged.

**Pull request merged**
